types/chrome/index.d.ts

```diff
@@ -5394,6 +5394,91 @@ declare namespace chrome.input.ime {
+
+    /**
+     * Resets the app instance identifier and revokes all tokens associated with it.
+     *
+     * The `deleteID()` method doesn't return any value, but can be used with a callback or asynchronously,
+     * with a Promise (MV3 only).
+     */
+    export function deleteID(): Promise<void>;
+    export function deleteID(callback: () => void): void;
+
+    interface DeleteTokenParams {
+        /** The authorized entity that is used to obtain the token. */
+        authorizedEntity: string;
+        /** The scope that is used to obtain the token. */
+        scope: string;
+    }
+    /**
+     * Revoked a granted token.
+     * @param deleteTokenParams.authorizedEntity The authorized entity that is used to obtain the token.
+     * @param deleteTokenParams.scope The scope that is used to obtain the token.
+     *
+     * The `deleteToken()` method doesn't return any value, but can be used with a callback or
+     * asynchronously, with a Promise (MV3 only).
+     */
+    export function deleteToken(deleteTokenParams: DeleteTokenParams): Promise<void>;
+    export function deleteToken(
+        deleteTokenParams: DeleteTokenParams,
+        callback: () => void,
+    ): void;
+
+    /**
+     * Retrieves the time when the InstanceID has been generated.
+     * @return The time when the Instance ID has been generated, represented in milliseconds since the epoch.
+     * It can return via a callback or asynchronously, with a Promise (MV3 only).
+     */
+    export function getCreationTime(): Promise<number>;
+    export function getCreationTime(callback: (creationTime: number) => void): void;
+
+    /**
+     * Retrieves an identifier for the app instance.
+     * The same ID will be returned as long as the application identity has not been revoked or expired.
+     *
+     * @return An Instance ID assigned to the app instance. Can be returned by a callback or a Promise (MV3 only).
+     */
+    export function getID(): Promise<string>;
+    export function getID(callback: (instanceID: string) => void): void;
+
+    // TODO: can i make this an extension on DeleteTokenParams?
+    interface GetTokenParams {
+        /**
+         * Identifies the entity that is authorized to access resources associated with this Instance ID.
+         * It can be a project ID from Google developer console.
+         */
+        authorizedEntity: string;
+        /**
+         * Allows including a small number of string key/value pairs that will be associated with the token
+         * and may be used in processing the request.
+         *
+         * @deprecated Since Chrome 89. `options` are deprecated and will be ignored.
+         */
+        options?: { [key: string]: string };
+        /**
+         * Identifies authorized actions that the authorized entity can take.
+         * E.g. for sending GCM messages, GCM scope should be used.
+         */
+        scope: string;
+    }
```

Comment:
You should be able to extend `DeleteTokenParams` like this. Can you change this or remove the comment?

```suggestion
    interface GetTokenParams extends DeleteTokenParams {
        /**
         * Allows including a small number of string key/value pairs that will be associated with the token
         * and may be used in processing the request.
         *
         * @deprecated Since Chrome 89. `options` are deprecated and will be ignored.
         */
        options?: { [key: string]: string };
    }
```

Reply:
Merging the comments on `authorizedEntity` was pretty easy, but to me, the longer comment on `scope` made less sense to me. However, I don't use/understand instanceID. I took the liberty of merging the two together, just wondering if you thought the new comment on `scope` made sense?

Reply:
I think the wording works as it is. If you wanted to be completely accurate you could change `DeleteTemplateParams.scope`'s comment to "The scope that was used to obtain the token.", then change the comment of `GetTokenParams.scope` to use the comment that's currently on `DeleteTokenParams` like this:

```suggestion
        options?: { [key: string]: string };
        /**
         * Identifies authorized actions that the authorized entity can take.
         * In other words, the scope that is used to obtain the token.
         * E.g. for sending GCM messages, `GCM` scope should be used.
         */
        scope: DeleteTokenParams['scope'];
```

That being said it looks good to me, and the above might be overkill. I also don't use GCM / InstanceID, though.  If you want to change it just update the pull request and I'll re-review it.